packages/snap-toolbox/src/until/until.ts

```diff
@@ -0,0 +1,65 @@
+export type UntilOptions = {
+	checkMax: number;
+	checkCount: number;
+	checkTime: number;
+	exponential: number;
+	defer: boolean;
+	executeFunction: boolean;
+};
+
+const defaultUntilOptions: UntilOptions = {
+	checkMax: 20,
+	checkCount: 0,
+	checkTime: 50,
+	exponential: 1.1,
+	defer: false,
+	executeFunction: true,
```

Comment:
Added this new `executeFunction` option since by default any function provided will execute before resolving without any function parameters. When this option is `false`, a reference is returned instead.

Reply:
I am slightly confused about this change. The main purpose of being able to use a function is for the function to be executed on the check interval and we would wait until said function returns true. This new option does not make sense to me... if you use it, and provided a function for `thing` it would always be truthy (as a function is) and thus would execute as expected.

If you were simply waiting for a variable that is expected to be a function to exist you could write a function to check for that, wait until that is true and then run the function.